packages/material-ui/src/Alert/Alert.js

```diff
@@ -167,42 +155,34 @@ const Alert = React.forwardRef(function Alert(props, ref) {
     ...other
   } = props;
 
-  const themeVariantsClasses = useThemeVariants(
-    {
-      ...props,
-      closeText,
-      iconMapping,
-      role,
-      severity,
-      variant,
-    },
-    'MuiAlert',
-  );
+  const styleProps = {
+    ...other,
+    variant,
+    color,
+    severity,
+  };
+
+  const classes = useUtilityClasses(styleProps);
 
   return (
-    <Paper
+    <AlertRoot
       role={role}
       square
       elevation={0}
-      className={clsx(
-        classes.root,
-        classes[variant],
-        classes[`${variant}${capitalize(color || severity)}`],
-        themeVariantsClasses,
-        className,
-      )}
+      styleProps={styleProps}
+      className={clsx(classes.root, className)}
       ref={ref}
       {...other}
     >
       {icon !== false ? (
-        <div className={classes.icon}>
+        <AlertIcon className={classes.icon}>
```

Comment:
```suggestion
        <AlertIcon className={classes.icon} styleProps= {styleProps}>
```

Same on the icon and action.